crates/matrix-sdk-indexeddb/src/crypto_store.rs

```diff
@@ -1212,6 +1274,42 @@ impl Drop for IndexeddbCryptoStore {
     }
 }
 
+/// The objects we store in the gossip_requests indexeddb object store
+#[derive(Debug, serde::Serialize, serde::Deserialize)]
+struct GossipRequestIndexedDbObject {
+    /// encrypted hash of the SecretInfo structure
+    info: String,
+
+    /// encrypted serialised representation of the GossipRequest as a whole
+    request: Vec<u8>,
+
+    /// whether the request has yet to be sent out
+    //
+    // Booleans don't really do what you might hope in indexeddb. Instead, we use an `Option`, and
+    // omit the value altogether if it is `None`, which means it will be excluded from the
+    // "unsent" index. If it is `Some`, the actual value is unimportant.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    unsent: Option<u8>,
```

Comment:
It is possible to have `Option<bool>` instead of `Option<u8>`? What happens if I pass `Some(7)`?

The doc says: "the actual value is unimportant”. So… maybe `Option` isn't the correct type here? If there is no choice, maybe use `Option<()>` instead of `Option<u8>` please.

If it's a matter of how we serialize/deserialize the value, maybe:

```rust
#[serde(skip_serializing_if = "is_true")]
unset: bool,
```

with:

```rust
fn is_true(b: &bool) -> bool {
    *b
}
```

should work (not tested).

The doc doesn't specify why booleans “don't really do what [we] might hope in indexeddb”. Why? Can you provide a source for this claim please? I would like to learn more about this :-).

Reply:
see comments at https://github.com/matrix-org/matrix-rust-sdk/pull/2626/files#r1338374673.